<launch>
	<node pkg="tf" type="static_transform_publisher" name="tf_base2mid360" args="0 0 1.005 0 0 0 /base_link /mid360_frame 10" />
	<node pkg="tf" type="static_transform_publisher" name="tf_base2laser" args="0.075 0 0.49 0 0 0 /base_link /laser 10" />
	<node pkg="tf" type="static_transform_publisher" name="tf_base2gps" args="0.075 0 0.49 0 0 0 /base_link /gps 10" />


	<node pkg="ypspur_ros" type="ypspur_ros" name="ypspur_ros" respawn="true">
		<param name="port" value="/dev/ttyACM0" />
		<param name="param_file" value="/home/nkb/icart-middle.param" />
		<param name="publish_odom_tf" value="false" />
	</node>

  <!-- ジョイスティック入力設定 -->
  <arg name="joy_device" default="/dev/input/js0" />
  <arg name="joy_deadzone" default="0.05" />
  <arg name="joy_autorepeat_rate" default="20.0" />
  <arg name="joy_coalesce_interval" default="0.02" />

  <!-- cmd_velスケール調整 -->
  <arg name="linear_scale" default="1.2" />
  <arg name="angular_scale" default="1.5" />
  <arg name="enable_button" default="4" />
  <arg name="turbo_button" default="5" />
  <arg name="turbo_ratio" default="1.5" />

  <!-- joyノード（PSコントローラ用に/dev/input/js0を仮定） -->
  <node pkg="joy" type="joy_node" name="joy_node" output="screen">
    <param name="dev" value="$(arg joy_device)" />
    <param name="deadzone" value="$(arg joy_deadzone)" />
    <param name="autorepeat_rate" value="$(arg joy_autorepeat_rate)" />
    <param name="coalesce_interval" value="$(arg joy_coalesce_interval)" />
  </node>

  <!-- Joystick teleoperation ノード -->
  <node pkg="tc2025" type="joystick_teleop.py" name="joystick_teleop" output="screen">
    <param name="linear_axis" value="1" />
    <param name="angular_axis" value="0" />
    <param name="linear_y_axis" value="-1" />
    <param name="linear_scale" value="$(arg linear_scale)" />
    <param name="angular_scale" value="$(arg angular_scale)" />
    <param name="linear_y_scale" value="0.5" />
    <param name="linear_axis_invert" value="False" />
    <param name="angular_axis_invert" value="False" />
    <param name="deadzone" value="0.05" />
    <param name="enable_button" value="$(arg enable_button)" />
    <param name="turbo_button" value="$(arg turbo_button)" />
    <param name="turbo_ratio" value="$(arg turbo_ratio)" />
    <param name="timeout" value="0.5" />
    <param name="publish_rate" value="20.0" />
  </node>

  <!-- Waypoint input ノード -->
  <node pkg="tc2025" type="waypoint_input.py" name="waypoint_input" output="screen" launch-prefix="xterm -e" />

  <!-- ublox GPS -->
  <include file="$(find ublox_gps)/launch/ublox_device.launch">
    <arg name="node_name" value="ublox" />
    <arg name="param_file_dir" value="$(find ublox_gps)/config" />
    <arg name="param_file_name" value="zed_f9p_rtk" />
  </include>

	<!-- MID360 -->
	<!--user configure parameters for ros start-->
  <group ns="mid360">
		<arg name="lvx_file_path" default="livox_test.lvx"/>
		<arg name="bd_list" default="100000000000000"/>
		<arg name="xfer_format" default="0"/>
		<arg name="multi_topic" default="0"/>
		<arg name="data_src" default="0"/>
		<arg name="publish_freq" default="10.0"/>
		<arg name="output_type" default="0"/>
		<arg name="rviz_enable" default="false"/>
		<arg name="rosbag_enable" default="false"/>
		<arg name="cmdline_arg" default="$(arg bd_list)"/>
		<arg name="msg_frame_id" default="mid360_frame"/>
		<arg name="lidar_bag" default="true"/>
		<arg name="imu_bag" default="true"/>
		<!--user configure parameters for ros end-->

		<param name="xfer_format" value="$(arg xfer_format)"/>
		<param name="multi_topic" value="$(arg multi_topic)"/>
		<param name="data_src" value="$(arg data_src)"/>
		<param name="publish_freq" type="double" value="$(arg publish_freq)"/>
		<param name="output_data_type" value="$(arg output_type)"/>
		<param name="cmdline_str" type="string" value="$(arg bd_list)"/>
		<param name="cmdline_file_path" type="string" value="$(arg lvx_file_path)"/>
		<param name="user_config_path" type="string" value="$(find livox_ros_driver2)/config/MID360_config.json"/>
		<param name="frame_id" type="string" value="$(arg msg_frame_id)"/>
		<param name="enable_lidar_bag" type="bool" value="$(arg lidar_bag)"/>
		<param name="enable_imu_bag" type="bool" value="$(arg imu_bag)"/>

		<node name="livox_lidar_publisher2" pkg="livox_ros_driver2"
					type="livox_ros_driver2_node" required="true"
					output="screen" args="$(arg cmdline_arg)"/>
  </group>

</launch>
